modules/auxiliary/scanner/http/softing_sis_login.rb

```diff
@@ -0,0 +1,146 @@
+require 'metasploit/framework/credential_collection'
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::AuthBrute
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Softing Secure Integration Server Login Utility',
+        'Description' => %q{
+          This module will attempt to authenticate to a Softing Secure Integration Server.
+        },
+        'Author' => [ 'Imran E. Dawoodjee <imrandawoodjee.infosec[at]gmail.com>' ],
+        'License' => MSF_LICENSE,
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        },
+        'DefaultOptions' => {
+          'RPORT' => 8099,
+          'SSL' => false,
+          'SSLVersion' => 'TLS1'
+        }
+      )
+    )
+
+    deregister_options('PASSWORD_SPRAY')
+
+    # credentials are "admin:admin" by default
+    register_options(
+      [
+        OptString.new('USERNAME', [false, 'The username to specify for authentication. Not required if SIGNATURE is set.', 'admin']),
+        OptString.new('PASSWORD', [false, 'The password to specify for authentication. Not required if SIGNATURE is set.', 'admin'])
+      ]
+    )
+  end
+
+  def scanner(ip)
+    @scanner ||= lambda {
+      cred_collection = Metasploit::Framework::CredentialCollection.new(
+        blank_passwords: datastore['BLANK_PASSWORDS'],
+        pass_file: datastore['PASS_FILE'],
+        password: datastore['PASSWORD'],
+        user_file: datastore['USER_FILE'],
+        userpass_file: datastore['USERPASS_FILE'],
+        username: datastore['USERNAME'],
+        user_as_pass: datastore['USER_AS_PASS']
+      )
+
+      return Metasploit::Framework::LoginScanner::SoftingSIS.new(
+        configure_http_login_scanner(
+          host: ip,
+          port: datastore['RPORT'],
+          cred_details: cred_collection,
+          stop_on_success: datastore['STOP_ON_SUCCESS'],
+          bruteforce_speed: datastore['BRUTEFORCE_SPEED'],
+          connection_timeout: 5
+        )
+      )
+    }.call
+  end
+
+  def report_good_cred(ip, port, result)
+    service_name = datastore['SSL'] ? 'https' : 'http'
+
+    service_data = {
+      address: ip,
+      port: port,
+      service_name: service_name,
+      protocol: 'tcp',
+      workspace_id: myworkspace_id
+    }
+
+    credential_data = {
+      module_fullname: fullname,
+      origin_type: :service,
+      private_data: result.credential.private,
+      private_type: :password,
+      username: result.credential.public
+    }.merge(service_data)
+
+    login_data = {
+      core: create_credential(credential_data),
+      last_attempted_at: DateTime.now,
+      status: result.status,
+      proof: result.proof
+    }.merge(service_data)
+
+    create_credential_login(login_data)
```

Comment:
Not required but may be worth testing this refactor.

Since this module imports `Exploit::Remote::HttpClient` there are helpers that should make this simpler.
```suggestion
    service_data = { status: result.status }.merge(service_details)
    store_valid_credential(user: result.credential.public, private: result.credential.private, proof: result.proof, service_data: service_data )
```

Although for this second pattern I am not 100% sure the result is equivalent:
``` Suggestion
   login_data = {
      core: result.credential,
      last_attempted_at: DateTime.now,
      status: result.status,
      proof: result.proof
    }.merge(service_details)
    create_credential_login(login_data)
```

Reply:
The second one throws an error `KeyError key not found: :workspace_id` then proceeds to not save successful creds. The first one works well. I've implemented the first refactor in commit 2f08cf6c46407d6a70f9c194679490f3ec7ea951.